src/commands/admin/addcourse.ts

```diff
@@ -1,32 +1,53 @@
-import { Message, OverwriteResolvable, Guild, TextChannel } from 'discord.js';
+import { Message, OverwriteResolvable, Guild, TextChannel, ApplicationCommandPermissionData, CommandInteraction, ApplicationCommandOptionData } from 'discord.js';
 import { Course } from '@lib/types/Course';
-import { adminPerms } from '@lib/permissions';
+import { ADMIN_PERMS } from '@lib/permissions';
 import { DB, GUILDS, ROLES } from '@root/config';
 import { Command } from '@lib/types/Command';
+import { modifyRoleDD } from '@root/src/lib/utils';
 
 export default class extends Command {
 
 	description = 'Creates a courses category and adds all necessary channels/roles.';
+	extendedHelp = 'This command will create all channels in a courses category,' +
+	'add all course roles, and add the course to the database. ';
 	usage = '<course ID>';
 	runInDM = false;
-	aliases = ['addc', 'createcourse', 'createc'];
+	tempPermissions: ApplicationCommandPermissionData[] = [ADMIN_PERMS];
 
-	async permissions(msg: Message): Promise<boolean> {
-		return await adminPerms(msg);
-	}
+	options: ApplicationCommandOptionData[] = [{
+		name: 'course',
+		description: 'The course ID of the course to be added (ex: 108).',
+		type: 'STRING',
+		required: true
+	}]
+
+	async tempRun(interaction: CommandInteraction): Promise<void> {
+		interaction.reply('<a:loading:755121200929439745> working...');
 
-	async run(msg: Message, [course]: [string]): Promise<Message> {
-		const response = msg.channel.send('<a:loading:755121200929439745> working...');
+		const course = interaction.options.getString('course');
+		//	make sure course does not exist already
+		if (await interaction.client.mongo.collection(DB.COURSES).countDocuments({ name: course }) > 0) {
+			throw `${course} has already been registered as a course.`;
```

Comment:
We can make this return an interaction reply instead of a throw, perhaps

Reply:
I'd rather keep it a throw for consistency (in terms of the other commands), plus I'm almost certain I kept it as a throw for a reason (I just can't remember the reason...).

Reply:
I've made my error handlers just embed replies, but I suppose it's negligible. (the former allows for ephemeral replies which I'd prefer just so chats aren't unreasonably clogged up)

Reply:
I think the distinction is going to be that `throw`s notify *us* of problems while `interaction.reply(...)` notifies the user of a problem. In this case, the user is us so I don't think it matters too much anyway